backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -1230,6 +1232,38 @@ class SubmissionDatabaseService(
             newVersion
         }
     }
+
+    fun getFileId(accessionVersion: AccessionVersion, fileField: String, fileName: String): FileId? {
+        return transaction {
+            val sql = """
+            select
+                files_of_field ->> 'fileId' as file_id
+            from
+                sequence_entries_view,
+                jsonb_array_elements(processed_data -> 'files' -> ?) as files_of_field
+            where
+                accession = ?
+                and version = ?
+                and files_of_field ->> 'name' = ?;
+            """.trimIndent()
+            return@transaction exec(
+                sql,
+                listOf(
+                    Pair(TextColumnType(), fileField),
+                    Pair(TextColumnType(), accessionVersion.accession),
+                    Pair(LongColumnType(), accessionVersion.version),
+                    Pair(TextColumnType(), fileName),
+                ),
+                explicitStatementType = StatementType.SELECT,
+            ) { resultSet ->
+                if (resultSet.next()) {
+                    UUID.fromString(resultSet.getString("file_id"))
+                } else {
+                    null
+                }
+            }
+        }
```

Comment:
Have you considered to just get the `processed_data`  for the `AccessionVersion`, parse it in Kotlin, and then get the FileID from there? That would have the advantage of type-safe JSON access, as well as not having the column and table names as strings in there (no literal SQL) with very little performance impact.

Reply:
This was for me the simplest and most compact/readable way to implement it and it is probably the most efficient. But I also won't mind a different solution.

Reply:
I've changed it in this PR: 

https://github.com/loculus-project/loculus/pull/3983/commits/b62f201839607b5be749c89919f80d66b16bdcda